### ISSUE TITLE
400 Bad Request on installation

### ISSUE THREAD

**Issue by wautersj** [n370/npm-publish-slack-webhook#1]
I'm afraid i'm not able to get it to work properly, it goes wrong starting from the installation.

I've tried installing it without the url at the end, and still configuring the script in 'postpublish', but no success so far.

<img width="921" alt="Screenshot 2021-01-27 at 22 55 05" src="https://user-images.githubusercontent.com/4070167/106059520-1c1e4f80-60f3-11eb-811f-07250279b452.png">

**Comment by n370** [comment]
Hey @wautersj thanks for trying it out. Let me check what is going on. I'll ping you back soon.

**Comment by n370** [comment]
Okay, so here is the thing. Apparently what you were trying to do was to install the library and at the same time use it. That's not how things work.

You have two options here.

First you can test the library and run it standalone with `npx`.

```
npx @n370/npm-publish-slack-webhook <your-slack-webhook-url>
```

If things run fine and you get the message on your channel, then you can install the library as a dependency to your project and configure your `postpublish` script to run it. If you do so, I advise you to *NOT* add the webhook url to your project's `package.json` file. Instead, set a new environment variable with the webhook url as its value and use it in your script  like:

```json
{
    "scripts": {
        "postpublish": "npm-publish-slack-webhook $MY_SLACK_WEBHOOK_URL"
    }
}
```

**Comment by wautersj** [comment]
I'm puzzled, I think I was doing exactly what was described :)
If I can get it to work / understand how it works, I can help improve the documentation of installation and usage.

![Screenshot 2021-01-28 at 09 47 04](https://user-images.githubusercontent.com/4070167/106112725-e6f31b00-614d-11eb-9b2f-d18bb325f9b8.png)

I've tried the npx command with the webhookurl, and no results, i'm afraid. 
(After I enter, i don't get any errors, but also. no slack message)

About the ENV variable, I understand that this is a best practice, but it's not important to get it to work. It should work directly with the webhook-url in the postpublish script.

**Comment by n370** [comment]
Hey @wautersj, after spending a little bit of time with this I found the issue. There was actually two.

First the readme content was incorrect, you were right.

Then you wasn't getting the notifications because I had a small logic issue in the source code where I was issuing a request with the http client to port 80, instead of using the https client against port 443.

https://github.com/n370/npm-publish-slack-webhook/commit/bbc40eb8af7205902070baa0fb902d8855716f4b#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L23-R23

All of those are fixed now and I published a new version 0.1.1 for you to give it a shot.

Please let me know how it goes so I can close this ticket. You're probably my first user so thank you for trying it out and also for reporting this issue.

**Comment by n370** [comment]
@wautersj I'm closing this issue for now. Please let me know if you need further assistance.
